### PR TITLE
Authentication part completed🚀

### PR DESCRIPTION
**Pull request by aayushjain-code**

### CODE REVIEW

File: src/Components/Authentication/ReqResAuth/RestrictAuth.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import { useLocation, Outlet, Navigate } from 'react-router-dom'
+import { useAuth } from "../../../context/authContext";
+
+
+const RistrictAuth = () => {
+	const { authState: { userId } } = useAuth();
+	console.log(userId)
```
Comment (hsnice16) [inline src/Components/Authentication/ReqResAuth/RestrictAuth.js]: I think, you can remove this console statement.

**Review by hsnice16 — APPROVED** [review]
Good work, nothing more to suggest.

File: src/backend/db/users.js [hunk 2]
```diff
@@ -12,7 +13,27 @@ export const users = [
...
+    updatedAt: formatDate(),
+  },
+
+  {
+    _id: uuid(),
+    firstName: "Adarsh",
+    lastName: "Balak",
+    email: 'test123@gmail.com',
+    password: bcyrpt.hashSync('testLogin@123', 5),
+    // password: 'testLogin@123',
```
Comment (Maianki) [inline src/backend/db/users.js:36]: You can remove commented code.

File: src/Components/Authentication/authentication.css [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	text-decoration: none;
+}
+
+form .text h3 a:hover {
+	text-decoration: underline;
+}
+
+
+
+
```
Comment (Maianki) [inline src/Components/Authentication/authentication.css:94]: Your code is not formatted you can use extensions like prettier to do that.

File: src/context/authContext.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			return state;
+	}
+}
+
+const AuthProvider = ({ children }) => {
+
+	const [authState, authDispatch] = useReducer(authReducer, initialState);
+
+	const signup = async (userDetails) => {
+		console.log('hello')
```
Comment (Maianki) [inline src/context/authContext.js:45]: Remove console.log() from the code.

**Review by Maianki — APPROVED** [review]
Good work Aayush. You have used good naming conventions for class names and components names. So, good work there. 
Things where you can work-
Right now your code is not formatted well so please use prettier or any of its alternative to format your code. It will make your code look more readable.

File: src/Components/Authentication/Signup.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+				</div>
+				<div className="input-box">
+					{toggleShowPassword ? <i class="fas fa-eye icon" onClick={() => setToggleShowPassword(!toggleShowPassword)}></i> : <i class="fas fa-eye-slash icon" onClick={() => setToggleShowPassword(!toggleShowPassword)}></i>}
+					<input type={toggleShowPassword ? "password" : 'text'} value={userDetails.password} onChange={(e) => setUserDetails({ ...userDetails, password: e.target.value })} placeholder="Create password" ></input>
+				</div>
+				<div className="input-box">
+					{toggleShowConfirmPass ? <i class="fas fa-eye icon" onClick={() => setToggleShowConfirmPass(!toggleShowConfirmPass)}></i> : <i class="fas fa-eye-slash icon" onClick={() => setToggleShowConfirmPass(!toggleShowConfirmPass)}></i>}
+					<input value={userDetails.confirmPassword} onChange={(e) => setUserDetails({ ...userDetails, confirmPassword: e.target.value })} type={toggleShowConfirmPass ? 'password' : 'text'} placeholder="Confirm password" ></input>
+				</div>
+				{/* <div className="policy">
```
Comment (MuskanChhatrasal) [inline src/Components/Authentication/Signup.js:83]: Remove commented code.

**Review by MuskanChhatrasal — APPROVED** [review]
Good to merge

File: src/Components/Authentication/ReqResAuth/RestrictAuth.js (hunk 1 above)
Comment (aayushjain-code) [reply src/Components/Authentication/ReqResAuth/RestrictAuth.js]: Will do it!

File: src/backend/db/users.js (hunk 2 above)
Comment (aayushjain-code) [reply src/backend/db/users.js:36]: Sure will implement it!

File: src/context/authContext.js (hunk 4 above)
Comment (aayushjain-code) [reply src/context/authContext.js:45]: Sure thanks

File: src/Components/Authentication/Signup.js (hunk 5 above)
Comment (aayushjain-code) [reply src/Components/Authentication/Signup.js:83]: okay

**Pull request merged**